### PR TITLE
feat: 1141 ownership cypress test

### LINKED ISSUE #1141: Ownership Cypress Test
1. Interact with the step 2 ownership of the A Class Registration form and come up with a test plan that tests all the functionalities of the form. Add the test plan as a comment in this issue. 
2. Implement tests according to the test plan.

Should test with at least 3 owner agencies

Use the same seedlot number for this test as the one from step 1 & 3's test

Definition of done:
- [x] Test plan created
- [x] Test implemented

### PR DESCRIPTION
**Pull request by ngunner15**
# Description
Testing this step - 
image
Closes #1141

#### New
- cypress test for Ownership(Step 2) of seedlot form

#### Changed
- cypress.config
- aclass-reg-form fixture(added ownership fixture)

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [x] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-4-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1204-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1204-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      .click();
+
+    cy.get('@ownerAgencySection')
+      .should('have.length', 1);
+  });
+
+  it('check owner portion % for 3 owner agencies', () => {
+    cy.get('button.owner-add-btn')
+      .click()
+      .blur();
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:263]: is `.blur()` needed for button click?

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      .find('.item-title-section')
+      .should('have.text', regFormData.agencyTitle);
+
+    cy.get(`.${prefix}--accordion__title`)
+      .eq(1)
+      .find('.item-description-section')
+      .should('have.text', regFormData.agencySubtitle);
+
+    cy.get('#ownership-portion-0')
+      .blur();
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:286]: what's this for 🤔

**Review by craigyu — COMMENTED** [review]
I know it's in draft mode, leaving a couple comments here just in case, please ignore if they are irrelevant

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts (hunk 2 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:286]: To get the error msg after adding 3 sections of owner agencies.

Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts]: https://github.com/bcgov/nr-spar/pull/1204/commits/4b573f97dc2e1445b285bd148008bbc7e0371329

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts (hunk 1 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts]: removed blur - https://github.com/bcgov/nr-spar/pull/1204/commits/4572409fb5e8cccb16dd7c8a5dd234771271ad2c

**Review by craigyu — CHANGES_REQUESTED** [review]
Good job Nisarg! 
As discussed, we should add one more test to check if the accordion title will change if the an owner's agency changes

**Review by mgaseta — APPROVED** [review]
Besides Craig's comments, LGTM!

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+
+    // Check accordian subtitle reflects change in ownership portion
+    cy.get('#ownership-portion-0')
+      .clear()
+      .type('80', { delay: TYPE_DELAY })
+      .blur();
+
+    cy.get(`.${prefix}--accordion__title`)
+      .find('.item-description-section')
+      .should('have.text', '80% owner portion');
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:248]: This addition is great! but what about the title with the agency acronym, they change when agency is cleared and filled, there will be two cases to cover
1. Empty agency selection should display "... some values" on the accordion title,
2. With an agency selection should "dislay "... some value with the agency acronym" on the accordion title

**Review by craigyu — CHANGES_REQUESTED** [review]

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+    // Check default title and subtitle created
+    cy.get(`.${prefix}--accordion__title`)
+      .eq(1)
+      .find('.item-title-section')
+      .should('have.text', regFormData.ownership.accordionTitle);
+
+    cy.get(`.${prefix}--accordion__title`)
+      .eq(1)
+      .find('.item-description-section')
+      .should('have.text', regFormData.ownership.accordionSubtitle);
```
Comment (ngunner15) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:401]: @craigyu You can see the empty agency selection test here

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+      .find('h3')
+      .should('have.text', regFormData.ownership.title);
+
+    cy.get('.ownership-header')
+      .find('p')
+      .should('have.text', regFormData.ownership.subtitle);
+
+    cy.get(`.${prefix}--accordion__title`)
+      .find('.item-title-section')
+      .should('have.text', initialAccordionTitle);
```
Comment (ngunner15) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:75]: @craigyu You can see an agency selection should "display "... some value with the agency acronym" on the accordion title

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts (hunk 3 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:248]: Those tests are in another it block

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts (hunk 5 above)
Comment (craigyu) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-ownership.cy.ts:75]: nice!

**Review by craigyu — APPROVED** [review]

**Review by RMCampos — APPROVED** [review]
After updating this branch code with `main`'s branch code, it works! Great job 💯

**Pull request merged**